xontrib/cmd_done.py

```diff
@@ -89,21 +89,30 @@ def _linux_is_app_window_focused():
 
 
 def _darwin_is_app_window_focused():
-    term = xsh.env.get("TERM_PROGRAM")
-    if not term:
+    appname = None
+
+    if bundle_id := xsh.env.get("__CFBundleIdentifier") :
+        out = sp.check_output(["lsappinfo", "find", "bundleID=" + bundle_id]).strip()
```

Comment:
if bundleid is available then we can run that directly like below to determine the result

`lsappinfo info -bundleid $__CFBundleIdentifier`